File: 1894-merge-strings-alternately/merge-strings-alternately.py
class Solution(object):
    def mergeAlternately(self, word1, word2):
        """
        :type word1: str
        :type word2: str
        :rtype: str
        """
        d= min(len(word1),len(word2))
        result = []
        i = 0

        while i < d:
            result.append(word1[i])
            result.append(word2[i])
            i+=1
        result+= word1[i:]
        result+= word2[i:]

        return ''.join(result)
        
        

File: 1894-merge-strings-alternately/merge-strings-alternately.java
class Solution {
    public String mergeAlternately(String word1, String word2) {
        StringBuilder st = new StringBuilder();
        int i = 0;
        boolean cont = true, s1, s2;
        while(cont){
            s1 = false;
            s2 = false;
            if (i<word1.length()){
                s1 = true;
                st.append(word1.charAt(i));
            }
            if (i<word2.length()){
                s2 = true;
                st.append(word2.charAt(i));
            }
            cont = s1 || s2;
            i++;
        }

        return st.toString();
    }
}